### ISSUE TITLE
[bug] CMakeToolchain/CMakeDeps generates CMake warnings when dependency was already found via find_package

### ISSUE THREAD

**Issue by n-morales** [conan-io/conan#10132]
### Environment Details (include every applicable attribute)
  * Operating System+version: Linux Mint 20.1 Ulyssa
  * Compiler+version: gcc 11.1.0
  * Conan version: 1.43.0
  * Python version: 3.8.10

### Steps to reproduce (Include if Applicable)
Use the CMakeToolchain and CMakeDeps generator.
Use the generated conan_toolchain.cmake file then `find_package` for a dependency and then `find_package` for a transitive dependency. E.g.
```cmake
find_package(PNG 1.6 REQUIRED)
find_package(Freetype 2.10 REQUIRED)
```

In this example, Freetype depends on PNG.

This could possibly be avoided by omitting the `find_package` for `PNG` (in this case), but that's not a good solution in case the consumer depends on `PNG` (or other transitively dependent library) and can't depend on the library being included transitively.

Also, adding `QUIET` to the `find_package` calls does not suppress the warning because they are emitted directly via `message(WARNING)` and so cannot be suppressed with that flag.

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
CMake Warning at cmake-build-debug/zlibTargets.cmake:24 (message):
  Target name 'zlib::zlib' already exists.
Call Stack (most recent call first):
  cmake-build-debug/zlib-config.cmake:11 (include)
  [omitted path to cmake]/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  cmake-build-debug/libpng-config.cmake:16 (find_dependency)
  [omitted path to cmake]/cmake-3.21/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  cmake-build-debug/freetype-config.cmake:16 (find_dependency)
  CMakeLists.txt:83 (find_package)
  ```

**Comment by memsharded** [comment]
I think the issue is doing a ``find_package()`` for a transitive dependency not directly declared in your ``conanfile.py``.
Conan 2.0 is going to completely isolate from transitive dependencies, that ``find_package()`` will probably fail because the headers will not be visible. This feature has been requested many times in the past, because it is necessary to maintain good architecture and good practices among developers. If you need to directly use PNG in your code, you need to add a ``requires="png..."`` in your recipe. 

Please try that, adding PNG as ``requires``, and the warnings should probably go away, please let us know.

**Comment by n-morales** [comment]
Hmm..., my recipe already has png -- sorry I didn't post it earlier:

```python
requires = [
        "fmt/8.0.1",
        "vulkan-loader/1.2.198.0",
        "opengl/system",
        "glslang/8.13.3559",
        "freetype/2.10.4",
        "pugixml/1.11",
        "spirv-cross/20210115",
        "zlib/1.2.11",
        "libpng/1.6.37",
# additional dependencies...
    ]
```

I tried re-ordering them, just in case it would matter but there was no change.

Looking at the generated cmake, it seems pretty apparent why the warning is throwing:

```cmake
if(NOT TARGET zlib::zlib)
    add_library(zlib::zlib INTERFACE IMPORTED)
    conan_message(STATUS "Conan: Target declared 'zlib::zlib'")
else()
    message(WARNING "Target name 'zlib::zlib' already exists.")
endif()
```

Since `zlib::zlib` was already generated by a previous `find_package` from `PNG` the second time it's called it will execute the else. I think the correct behavior would be to just not emit a warning. I believe normal behavior for cmake `find_package` is to just silently re-use the previously found package, so omitting the warning would bring the behavior closer to standard cmake.

**Comment by memsharded** [comment]
@franramirez688 you were checking last time about this warning with @lasote, do you see any problem in omitting the warning?

**Comment by lasote** [comment]
Yeah, let's remove it. We didn't realize about this use case.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10150 removed it, will be released in 1.44

**Comment by magist3r** [comment]
@memsharded I still have similar warnings for ALIAS targets declared with `self.cpp_info.set_property("cmake_target_aliases", ["my_target::my_target"])`.

For example:
```
CMake Warning at module-my_targetTargets.cmake:25 (message):
  Target name 'my_target::my_target' already exists.
```

Conan 1.48. 
I think we can get rid of these warnings too.
